helm/policies-shared/tests/abs/test_shared_default.py

```diff
@@ -26,9 +27,22 @@ def test_silence_heartbeat_policy(silence) -> None:
     """
     matchers = silence['spec']['matchers']
 
-    for i, v in enumerate(matchers):
-        if v['name'] == "alertname":
-            assert (v['value'] == "Heartbeat" and not v['isRegex'] and not v['isEqual'])
+    for matcher in matchers:
+        if matcher['name'] == "alertname":
+            assert (matcher['value'] == "Heartbeat" and not matcher['isRegex'] and not matcher['isEqual'])
             return
 
     pytest.fail("Heartbeat matcher is missing")
+
+
+@pytest.mark.smoke
+def test_service_monitor_labelling_schema_policy(servicemonitor) -> None:
+    """
+    test_service_monitor_labelling_schema_policy tests defaulting of an empty Service monitor to check that the labelling schema is configured.
+
+    :param servicemonitor: Any ServiceMonitor CR.
+    """
+    endpoints = servicemonitor['spec']['endpoints']
+    for endpoint in endpoints:
+        relabelings = endpoint['relabelings']
+        assert len(relabelings) > 0, 'Invalid relabelings {}'.format(relabelings)
```

Comment:
I am not sure why you assert over the length here - what does this show exactly?

Reply:
It ensures that the relabelings are added as the service monitor is default by default but it might make sense to test the values.

I also tried to work on a validating webhook but I could not make it work :(

Reply:
Yes verifying some of the values might be nice to make sure that we are actually defaulting correctly :)

Reply:
It should be fine now :) Would you have a bit of time to help with the validating webhook or know someone that could help ?

Reply:
FYI, this is what i'm trying : 
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: validate-service-monitor-labelling-schema
spec:
  validationFailureAction: enforce
  rules:
    - name: validate-service-monitor-app-label
      match:
        resources:
          kinds:
          - monitoring.coreos.com/v1/ServiceMonitor
      validate:
        message: "You must define the app label on all new service monitor endpoints."
        pattern:
          spec:
            endpoints:
            - ^(relabelings):
              - targetLabel: "app"
#    - name: validate-service-monitor-cluster-id-label
#      match:
#        resources:
#          kinds:
#          - monitoring.coreos.com/v1/ServiceMonitor
#      validate:
#        message: "You must define the cluster id label on all new service monitor endpoints."
#        pattern:
#          spec:
#            endpoints:
#            - ^(relabelings):
#              - =(targetLabel): "cluster_id"
#    - name: validate-service-monitor-cluster-type-label
#      match:
#        resources:
#          kinds:
#          - monitoring.coreos.com/v1/ServiceMonitor
#      validate:
#        message: "You must define the cluster type label on all new service monitor endpoints."
#        pattern:
#          spec:
#            endpoints:
#            - port: "*"
#              ^(relabelings):
#              - =(targetLabel): "cluster_type"

Reply:
Can you post this in them team-ludacris slack? The markdown here is all messed up and then I can give it a closer look tomorrow